test/new-e2e/examples/vm_fakeintake_test.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package examples
+
+import (
+	_ "embed"
+	"errors"
+	"time"
+
+	"testing"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/utils/e2e"
+	commonos "github.com/DataDog/test-infra-definitions/components/os"
+	ec2vm "github.com/DataDog/test-infra-definitions/scenarios/aws/vm/ec2VM"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/vm/os"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type vmFakeintakeSuite struct {
+	e2e.Suite[e2e.AgentEnv]
+}
+
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &vmFakeintakeSuite{}, e2e.AgentStackDef([]e2e.Ec2VMOption{ec2vm.WithArch(os.UbuntuOS, commonos.AMD64Arch)}))
+}
+
+func (s *vmFakeintakeSuite) TestVM() {
+	output := s.Env().VM.Execute("ls")
+	require.NotEmpty(s.T(), output)
+}
+
+func (s *vmFakeintakeSuite) TestAgent() {
+	err := s.Env().Agent.WaitForReady()
```

Comment:
Note: When reading this code I have the feeling that I should always call `WaitForReady()`. We should consider hiding `WaitForReady` and call it inside Status in a followup PR.

Reply:
Interesting. I would not call `WaitForReady` inside `Status` as it would hide it, and some tests might require to skip waiting. I consider devs agent-savvy enough to know that the agent is not ready right away, and that they need the agent should be ready before starting asking for the agent status. Some tests might not need to call `Status` and call another command, but `WaitForReady` is still required as first command in a normal flow. To be discussed !

Reply:
I still need to merge fixes from this PR